### PR TITLE
users unable to search files

### PR DESCRIPTION
**Pull request by tcnichol**
I change the name here since the bug affected more than just read only users.

If a user who is not the owner of a dataset searched for files, they would not find files in the dataset.

This is because the user_ids were not updated in the elasticsearch entries for the files. 

To fix this, I added a method index_dataset_files which is called after datasets are updated and indexed.

I also checked and a user that is part of a group will find files in the group. Ready for review.

### CODE REVIEW

**Comment by tcnichol** [comment]
I'm converting to draft again because some issues are coming up with groups. Adding and removing members from groups seems to mean that some users who are in a group that has access to a file or dataset will not get it in search results.

**Comment by tcnichol** [comment]
Important Note:

I cannot get the shellcheck pre-commit hook to run on my machine. So far I have found no solution - I can install it via brew but if my pre-commit config is not the same as the one in the github repo, it rejects the commit. 

So if someone else can please run it as part of the review, thank you very much. 

This should be ready to review now.

File: backend/app/routers/datasets.py [hunk 1]
```diff
@@ -783,6 +783,7 @@ async def save_files(
                 authenticated=authenticated,
             )
             files_added.append(new_file.dict())
+            await index_file(es, FileOut(**new_file.dict()), update=True)
```
Comment (longshuicy) [inline backend/app/routers/datasets.py:786]: Duplicated. In the add_file_entry() function above, files are already indexed.

```
# Add entry to the file index
    await index_file(es, FileOut(**new_file.dict()))
```

File: backend/app/routers/groups.py [hunk 2]
```diff
@@ -189,6 +192,11 @@ async def edit_group(
                 ).update(
                     Push({AuthorizationDB.user_ids: user.email}),
                 )
+            group_authorizations = await AuthorizationDB.find(
+                AuthorizationDB.group_ids == ObjectId(group_id)
+            ).to_list()
+            for auth in group_authorizations:
+                print("add auth")
```
[the remark this answers is not in the document]
Comment (tcnichol) [reply backend/app/routers/groups.py:199]: Removed.

File: backend/app/search/index.py [hunk 3]
```diff
@@ -114,6 +114,18 @@ async def index_file(
...
+async def index_dataset_files(
+    es: Elasticsearch,
+    dataset_id: str,
+):
+    query = [
+        FileDB.dataset_id == ObjectId(dataset_id),
+    ]
+    files = await FileDB.find(*query).to_list()
+    for file in files:
+        await index_file(es, FileOut(**file.dict()), update=True)
```
[the remark this answers is not in the document]
Comment (tcnichol) [reply backend/app/search/index.py:126]: Changed. It how has update as parameter like the others, and I've changed when it's called to include the parameter.

**Comment by ddey2** [comment]
I see this. Could you please check? 
Screenshot 2024-06-24 at 1 49 42 PM

**Review by ddey2 — APPROVED** [review]
works well, code looks good.

**Review by longshuicy — APPROVED** [review]
Works now.

**Pull request merged**